charts/stable/otel-collector/templates/common.yaml

```diff
@@ -0,0 +1,20 @@
+{{/* Make sure all variables are set properly */}}
+{{- include "common.values.setup" . }}
+
+{{- define "otel-collector.configVolume" -}}
+{{/* Append the secret volume to the volumes */}}
+enabled: true
+mountPath: "/conf/otel-collector-config.yaml"
+subPath: "otelConfigFile"
+type: "custom"
+volumeSpec:
+  secret:
+    {{- if .Values.configFileSecret }}
+    secretName: "{{ .Values.configSecretName }}"
+    {{- else }}
+    secretName: "{{ include "common.names.fullname" . }}-otelconfig"
+    {{- end }}
+{{- end -}}
+{{- $_ := set .Values.persistence "otel-config-file" (include "otel-collector.configVolume" . | fromYaml) -}}
```

Comment:
Take a look at https://github.com/k8s-at-home/charts/blob/b1fc8dd547f6b9164e66824a661f0c2ff508ca9f/charts/stable/ghost/templates/common.yaml#L4L13 to see how we handle setting "hardcoded" values in our other charts.